add -W argument to ignore warnings

Randomneo opened the pull request:
sometimes you'd like to ignore warnings but there is no argument for it. WDYT

wbolster commented:
not opposed to the idea itself, but what should this do?

```
  --disable-warnings, --disable-pytest-warnings
                        Disable warnings summary
```

or

```
pytest-warnings:
  -W PYTHONWARNINGS, --pythonwarnings=PYTHONWARNINGS
                        Set which warnings to report, see -W option of Python
                        itself
```
?

also, ‘disable’ has a typo

Randomneo commented:
fixed typos. I assume it should do the second one `-W PYTHONWARNINGS`.

Should I change description from `disable warnings` to `ignore warnings`?

Also changed arg to make it more obvious

wbolster commented on python-pytest.el:
```diff
@@ -125,6 +125,7 @@ When non-nil only ‘test_foo()’ will match, and nothing else."
    [("-c" "color" "--color")
     ("-q" "quiet" "--quiet")
     ("-s" "no output capture" "--capture=no")
+    ("-w" "disable warnings" "-W ignore")
```
let's keep it `-W` (capital) so that it matches both the corresponding and `pytest` and `python` args (both also use `-W`)

wbolster replied:
```suggestion
    ("-W" "ignore warnings" "-W ignore")
```

wbolster commented on python-pytest.el:
```diff
@@ -125,6 +125,7 @@ When non-nil only ‘test_foo()’ will match, and nothing else."
    [("-c" "color" "--color")
     ("-q" "quiet" "--quiet")
     ("-s" "no output capture" "--capture=no")
+    ("-W" "ignore warnings" "-W ignore")
     (python-pytest:-v)]]
```
let's keep alphabetical ordering btw

```suggestion
    (python-pytest:-v)
    ("-W" "ignore warnings" "-W ignore")]]
```

wbolster commented:
thanks, merged with a few tiny tweaks! 🚀
